### PR TITLE
Consumable quality enum (and support missing qualities)

### PR DESCRIPTION
**Pull request by gausie**
* Change string quality to an enum
* Track drippy, ??? and quest quality all of which exist in KoL
* Add sushi pseudoquality - should we call this pseudo?
* Add some colouring for drippy (brown seemed right) and some visualisation for ??? (i.e. changing)

### CODE REVIEW

**Review by Veracity0 — APPROVED** [review]
This awesome.

I especially like the ListCellRendererFactory test; that is not the easiest code to get right, and having a test suite will be very useful.

File: src/net/sourceforge/kolmafia/persistence/DateTimeManager.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  public static ZonedDateTime getArizonaDateTime() {
+    return ZonedDateTime.now(ARIZONA);
+  }
+
+  public static ZonedDateTime getRolloverDateTime() {
+    return ZonedDateTime.now(ROLLOVER);
+  }
+
+  private DateTimeManager() {}
+  ;
```
Comment (midgleyc) [inline src/net/sourceforge/kolmafia/persistence/DateTimeManager.java:25]: can omit?

File: test/net/sourceforge/kolmafia/persistence/HolidayDatabaseTest.java [hunk 2]
```diff
@@ -19,15 +18,18 @@ class HolidayDatabaseTest {
   class HamburgularPosition {
     @Test
     void noPositionBeforeCollision() {
-      var date = ZonedDateTime.of(2006, Month.JUNE.getValue(), 2, 0, 0, 0, 0, ROLLOVER);
+      var date =
+          ZonedDateTime.of(2006, Month.JUNE.getValue(), 2, 0, 0, 0, 0, DateTimeManager.ROLLOVER);
```
Comment (midgleyc) [inline test/net/sourceforge/kolmafia/persistence/HolidayDatabaseTest.java:22]: I think you could use a static import as before to save everything going across many lines ;)

File: src/net/sourceforge/kolmafia/persistence/ConsumablesDatabase.java [hunk 3]
```diff
@@ -58,12 +59,53 @@ public class ConsumablesDatabase {
...
+    ConsumableQuality(final String name) {
+      this(name, null);
+    }
+
+    @Override
+    public String toString() {
+      return name;
+    }
+
+    public java.lang.String getName() {
```
[the remark this answers is not in the document]
Comment (gausie) [reply src/net/sourceforge/kolmafia/persistence/ConsumablesDatabase.java:101]: can!

File: test/net/sourceforge/kolmafia/persistence/HolidayDatabaseTest.java (hunk 2 above)
Comment (gausie) [reply test/net/sourceforge/kolmafia/persistence/HolidayDatabaseTest.java:22]: can!

File: src/net/sourceforge/kolmafia/persistence/DateTimeManager.java (hunk 1 above)
Comment (gausie) [reply src/net/sourceforge/kolmafia/persistence/DateTimeManager.java:25]: can!

**Review by midgleyc — APPROVED** [review]

**Pull request merged**